daemon/src/wallet.rs

```diff
@@ -345,17 +356,99 @@ where
     }
 }
 
+struct LockedUtxos {
+    inner: HashSet<(Instant, OutPoint)>,
+    time_to_lock: Duration,
+}
+
+impl LockedUtxos {
+    fn new(network: Network) -> Self {
+        let time_to_lock = match network {
+            Network::Bitcoin => Duration::from_secs(20 * 60),
+            Network::Testnet | Network::Signet | Network::Regtest => Duration::from_secs(2 * 60),
+        };
```

Comment:
👀 I didn't invest time into figuring out good values for these in case we decide to solve this whole issue differently. Suggestions are welcome.